misc/scripts/install-local.sh

```diff
@@ -125,6 +123,56 @@ function log() {
     } | sudo tee "$METADIR/$name" > /dev/null
 }
 
+function parse_source_entry() {
+    unset url dest git_branch git_tag git_commit
+    local entry="$1"
+    url="${entry#*::}"
+    dest="${entry%%::*}"
+    if [[ $entry != *::* && $entry == *#*=* ]]; then
+        dest="${url%%#*}"
+        dest="${dest##*/}"
+    fi
+    case $url in
+        *#branch=*)
+            git_branch="${url##*#branch=}"
+            url="${url%%#branch=*}"
+            ;;
+        *#tag=*)
+            git_tag="${url##*#tag=}"
+            url="${url%%#tag=*}"
+            ;;
+        *#commit=*)
+            git_commit="${url##*#commit=}"
+            url="${url%%#commit=*}"
+            ;;
+    esac
+    url="${url%%#*}"
+    if [[ $entry == *::* ]]; then
+        dest="${entry%%::*}"
+    elif [[ $entry != *#*=* ]]; then
+        url="$entry"
+        dest="${url##*/}"
+    fi
+}
+
+function calc_git_pkgver() {
+    unset comp_git_pkgver
+    local calc_commit
+    if [[ $url == git+* ]]; then
+        url="${url#git+}"
+    fi
+    if [[ -n ${git_branch} ]]; then
+        calc_commit="git ls-remote ${url} ${git_branch}"
+    elif [[ -n ${git_tag} ]]; then
+        calc_commit="git ls-remote ${url} ${git_tag}"
+    elif [[ -n ${git_commit} ]]; then
+        calc_commit="echo ${git_commit}"
+    else
+        calc_commit="git ls-remote ${url} HEAD"
+    fi
+    comp_git_pkgver="$(${calc_commit} | cut -f1 | cut -c1-8)"
```

Comment:
```suggestion
    if [[ -n ${git_branch} ]]; then
        : "git ls-remote ${url} ${git_branch}"
    elif [[ -n ${git_tag} ]]; then
        : "git ls-remote ${url} ${git_tag}"
    elif [[ -n ${git_commit} ]]; then
        : "echo ${git_commit}"
    else
        : "git ls-remote ${url} HEAD"
    fi
    comp_git_pkgver="$(${_} | cut -f1 | cut -c1-8)"
```

If you don't want to merge this that's fine, but this *would* lower character count.